mono/utils/atomic.h

```diff
@@ -347,130 +347,130 @@ static inline gint64 InterlockedRead64(volatile gint64 *src)
-extern gint32 InterlockedDecrement(volatile gint32 *dest);
-extern gint64 InterlockedDecrement64(volatile gint64 *dest);
-extern gint32 InterlockedExchange(volatile gint32 *dest, gint32 exch);
-extern gint64 InterlockedExchange64(volatile gint64 *dest, gint64 exch);
-extern gpointer InterlockedExchangePointer(volatile gpointer *dest, gpointer exch);
-extern gint32 InterlockedExchangeAdd(volatile gint32 *dest, gint32 add);
-extern gint64 InterlockedExchangeAdd64(volatile gint64 *dest, gint64 add);
-extern gint8 InterlockedRead8(volatile gint8 *src);
-extern gint16 InterlockedRead16(volatile gint16 *src);
-extern gint32 InterlockedRead(volatile gint32 *src);
-extern gint64 InterlockedRead64(volatile gint64 *src);
-extern gpointer InterlockedReadPointer(volatile gpointer *src);
-extern void InterlockedWrite8(volatile gint8 *dst, gint8 val);
-extern void InterlockedWrite16(volatile gint16 *dst, gint16 val);
-extern void InterlockedWrite(volatile gint32 *dst, gint32 val);
-extern void InterlockedWrite64(volatile gint64 *dst, gint64 val);
-extern void InterlockedWritePointer(volatile gpointer *dst, gpointer val);
+extern gint32 mono_atomic_cas_i32(volatile gint32 *dest, gint32 exch, gint32 comp);
+extern gint64 mono_atomic_cas_i64(volatile gint64 *dest, gint64 exch, gint64 comp);
+extern gpointer mono_atomic_cas_ptr(volatile gpointer *dest, gpointer exch, gpointer comp);
+extern gint32 mono_atomic_add_i32(volatile gint32 *dest, gint32 add);
+extern gint64 mono_atomic_add_i64(volatile gint64 *dest, gint64 add);
+extern gint32 mono_atomic_inc_i32(volatile gint32 *dest);
+extern gint64 mono_atomic_inc_i64(volatile gint64 *dest);
+extern gint32 mono_atomic_dec_i32(volatile gint32 *dest);
+extern gint64 mono_atomic_dec_i64(volatile gint64 *dest);
+extern gint32 mono_atomic_xchg_i32(volatile gint32 *dest, gint32 exch);
+extern gint64 mono_atomic_xchg_i64(volatile gint64 *dest, gint64 exch);
+extern gpointer mono_atomic_xchg_ptr(volatile gpointer *dest, gpointer exch);
+extern gint32 mono_atomic_xchg_i32Add(volatile gint32 *dest, gint32 add);
+extern gint64 mono_atomic_xchg_i32Add64(volatile gint64 *dest, gint64 add);
+extern gint8 mono_atomic_load_i8(volatile gint8 *src);
+extern gint16 mono_atomic_load_i16(volatile gint16 *src);
+extern gint32 mono_atomic_load_i32(volatile gint32 *src);
+extern gint64 mono_atomic_load_i64(volatile gint64 *src);
+extern gpointer mono_atomic_load_ptr(volatile gpointer *src);
+extern void mono_atomic_store_i8(volatile gint8 *dst, gint8 val);
+extern void mono_atomic_store_i16(volatile gint16 *dst, gint16 val);
+extern void mono_atomic_store_i32(volatile gint32 *dst, gint32 val);
+extern void mono_atomic_store_i64(volatile gint64 *dst, gint64 val);
+extern void mono_atomic_store_ptr(volatile gpointer *dst, gpointer val);
 
 #endif
 
 #if SIZEOF_VOID_P == 4
-#define InterlockedAddP(p,add) InterlockedAdd ((volatile gint32*)p, (gint32)add)
+#define mono_atomic_add_ptr(p,add) mono_atomic_add_i32 ((volatile gint32*)p, (gint32)add)
```

Comment:
This new name is kinda misleading. We have other `_ptr` functions which operate on actual pointers, but these macros operate on integers. This is an important distinction because e.g. `std::atomic` in C++ does support add/subtract operations on pointers. For example, `std::atomic<int *> x = ...; x.fetch_add (1);` will increment `x` by `sizeof (int)` just like on a regular pointer value. That's why I suggested the `_word` naming instead, which makes it clear (hopefully?) that the macros operate on integer values.